### PR TITLE
Upgrade to latest polkadot-sdk@1.6 release

### PR DESCRIPTION
**Pull request by bkontur**
Based on bump to [`polkadot-sdk@1.5.0`](https://github.com/polkadot-fellows/runtimes/pull/137).

Attached result of `cargo upgrade -v --pinned --incompatible` [cargo-upgrade-version-bump.log](https://github.com/polkadot-fellows/runtimes/files/14044160/cargo-upgrade-version-bump.log)

_Note: Encointer was not upgraded (because its pallet references `polkadot@1.3.0` release)._

## ~~For reviewers~~

~~This PR is against `polkadot-fellows`'s main to bring it to the fellows repo, but if you want to see a real diff relevant to the `polkadot-sdk@1.6.0` update please check: https://github.com/bkontur/runtimes/compare/bko-bump-to-1.5...bkontur:runtimes:bko-bump-to-1.6.~~

## TODO

- [x] fix compilation
- [x] fix integration tests
- [x] fix benchmarks (also try them) - `collectives-polkadot` `payout`
- [ ] ~~Does not require a CHANGELOG entry~~
- [x] `warning: use of deprecated struct `staging_xcm_builder::CurrencyAdapter`: Use `FungibleAdapter` instead`
  - [ ] search for `TODO:(PR#159) change to FungibleAdapter` and/or wait for https://github.com/paritytech/polkadot-sdk/pull/3287
- [x] patch for `pallet-nomination-pools` migration fix https://github.com/paritytech/polkadot-sdk/pull/3093
  - will be fixed here https://github.com/polkadot-fellows/runtimes/issues/188#issuecomment-1945556395
- [x] patch for `xcm-executor` fix (for 1.6.0) e.g. https://github.com/paritytech/polkadot-sdk/pull/3174
- [x] check/fix coretime stuff for Kusama/Polkadot - search for `// TODO:(PR#159)(PR#1694)` - see [comment](https://github.com/polkadot-fellows/runtimes/pull/159#issuecomment-1930163319) - fixed by https://github.com/bkontur/runtimes/pull/3
- [x] check the `MaxControllersInDeprecationBatch`https://github.com/polkadot-fellows/runtimes/pull/159/files#r1492361038
- [x] check `pallet_identity::Config` for Kusama and Polkadot https://github.com/polkadot-fellows/runtimes/pull/159/files#r1492363866

### CODE REVIEW

**Comment by bkontur** [comment]
Hey @eskimor, as a author of https://github.com/paritytech/polkadot-sdk/pull/1694,could you please check this commit
https://github.com/polkadot-fellows/runtimes/pull/159/commits/b8dd1b98660aa2dab78667682d14c450bd4ab562  (or search for `// TODO:(PR#159)(PR#1694)`) which is reflecting your #1694 from [polkadot-sdk@1.6.0](https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.6.0) release, and/or push directly to this branch: https://github.com/bkontur/runtimes/tree/bko-bump-to-1.6 or create PR to this branch (whatever is better for you)

File: relay/kusama/src/lib.rs
Comment (bkontur) [inline relay/kusama/src/lib.rs:687]: Hey @rossbulat, as an author of https://github.com/paritytech/polkadot-sdk/pull/2589, could you please confirm or adjust correct value of `MaxControllersInDeprecationBatch` for Kusama and Polkadot?

File: relay/kusama/src/lib.rs
Comment (bkontur) [inline relay/kusama/src/lib.rs:945]: @joepetrowski Joe, as an author of https://github.com/paritytech/polkadot-sdk/pull/2651, can you please check this identity stuff for Kusama and for Polkadot?

Comment (joepetrowski) [reply relay/kusama/src/lib.rs:945]: Looks good for both 👍

File: relay/kusama/src/lib.rs
Comment (bkontur) [inline relay/kusama/src/lib.rs:939]:
```suggestion
```

File: relay/polkadot/src/lib.rs
Comment (bkontur) [inline relay/polkadot/src/lib.rs:820]:
```suggestion
```

**Review by bkontur — COMMENTED** [review]
removed TODO based on the comment: https://github.com/polkadot-fellows/runtimes/pull/159/files#r1492363866

**Comment by bkontur** [comment]
Fellows, please review this bump.

Missing things:
- failed `check-migrations (Polkadot)` job, but this will be automatically resolved when `Polkadot@1.1.2` is enacted on February 19: [link to referendum](https://polkadot.polkassembly.io/referenda/458).
- TODOs for check coretime stuff  @antonva 
- TODOs for check `MaxControllersInDeprecationBatch` @rossbulat 
- test pipelines issue `out of space` - will be fixed in the separate PR: https://github.com/polkadot-fellows/runtimes/pull/189

File: relay/kusama/src/lib.rs
Comment (rossbulat) [inline relay/kusama/src/lib.rs:687]: There are 5596 unmatching stash / controller pairs on Kusama remaining.

```suggestion
	type MaxControllersInDeprecationBatch = ConstU32<5596>;
```

File: relay/polkadot/src/lib.rs
Comment (rossbulat) [inline relay/polkadot/src/lib.rs:781]: There are 5890 unmatching stash / controller pairs on Polkadot remaining.
```suggestion
	type MaxControllersInDeprecationBatch = ConstU32<5890>;
```

File: relay/polkadot/src/lib.rs
Comment (rossbulat) [inline relay/polkadot/src/lib.rs:780]:
```suggestion
```

File: relay/kusama/src/lib.rs
Comment (rossbulat) [inline relay/kusama/src/lib.rs:686]:
```suggestion
```

File: relay/kusama/src/lib.rs
Comment (rossbulat) [reply relay/kusama/src/lib.rs:687]: No problem @bkontur, I've left comments with the up to date figures. Thank you for flagging.

File: relay/kusama/src/lib.rs
Comment (muharem) [inline relay/kusama/src/lib.rs:686]: quite big, how we settled on this number?

File: relay/kusama/src/lib.rs
Comment (muharem) [inline relay/kusama/src/lib.rs:1227]: there are few new todos, are you doing to resolve them?

File: relay/polkadot/src/lib.rs
Comment (muharem) [inline relay/polkadot/src/lib.rs:821]: Only with root track? may be plus GeneralAdmin?

File: system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs
Comment (muharem) [inline system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:340]: changes to this config look right

File: relay/polkadot/src/lib.rs
Comment (joepetrowski) [reply relay/polkadot/src/lib.rs:821]: Optimistically the Kusama People Chain launch goes well and we do Polkadot one release cycle later. But using the feature will just mean more for UIs to migrate. Then yes I think it should be an easier origin for this.

File: relay/kusama/src/lib.rs
Comment (bkontur) [reply relay/kusama/src/lib.rs:686]: please see @rossbulat's [comment](https://github.com/polkadot-fellows/runtimes/pull/159#discussion_r1492649084)

File: relay/kusama/src/lib.rs
Comment (bkontur) [reply relay/kusama/src/lib.rs:1227]: those are the last open point from description and @antonva should be working on them see [comment](https://github.com/polkadot-fellows/runtimes/pull/159#issuecomment-1942479264)

**Comment by bkontur** [comment]
> Fellows, please review this bump.
> 
> Missing things:
> 
>     * [x]  failed `check-migrations (Polkadot)` job, but this will be automatically resolved when `Polkadot@1.1.2` is enacted on February 19: [link to referendum](https://polkadot.polkassembly.io/referenda/458).
> 
>     * [ ]  TODOs for check coretime stuff  @antonva
> 
>     * [x]  TODOs for check `MaxControllersInDeprecationBatch` @rossbulat
> 
>     * [x]  test pipelines issue `out of space` - will be fixed in the separate PR: [Split `--workspace` test pipeline to separate jobs #189](https://github.com/polkadot-fellows/runtimes/pull/189)

Finally, the CI is green including all the tests and check-migrations, so we are missing here just [coretime cfg](https://github.com/polkadot-fellows/runtimes/pull/159#issuecomment-1930163319) and more reviews

**Comment by brenzi** [comment]
I bumped the encointer pallets and upgraded the encointer runtime to 1.6.0. I'd suggest to merge my PR #193 into this one before the merge to have a consistent PR

File: relay/kusama/src/lib.rs
Comment (acatangiu) [inline relay/kusama/src/lib.rs:1306]: @antonva any progress here?

**Review by acatangiu — APPROVED** [review]

**Comment by brenzi** [comment]
@bkontur can we get this one in before merging? I think reviews would be simpler because the entire repo will have consistent dependencies:
https://github.com/bkontur/runtimes/pull/2

File: relay/kusama/src/lib.rs
Comment (bkontur) [reply relay/kusama/src/lib.rs]: fixed by: https://github.com/bkontur/runtimes/pull/3

**Review by ordian — APPROVED** [review]
Anything missing apart from changes from Encointer update to 1.6 to dedup deps?

**Comment by bkontur** [comment]
> Anything missing apart from changes from Encointer update to 1.6 to dedup deps?

@ordian not really, we just want to merge this [update](https://github.com/polkadot-fellows/runtimes/pull/185) to the xcm emulator tests here. The Encointer bump from @brenzi is ready here: https://github.com/bkontur/runtimes/pull/4, can be merged, but `check-migrations` for Encointer now fails - more investigation here: https://github.com/polkadot-fellows/runtimes/issues/200

**Comment by bkontur** [comment]
> > Anything missing apart from changes from Encointer update to 1.6 to dedup deps?
> 
> @ordian not really, we just want to merge this [update](https://github.com/polkadot-fellows/runtimes/pull/185) to the xcm emulator tests here. The Encointer bump from @brenzi is ready here: [bkontur#4](https://github.com/bkontur/runtimes/pull/4), can be merged, but `check-migrations` for Encointer now fails - more investigation here: #200

Xcm emulator tests are merged.
The Encointer bump will be merged here https://github.com/bkontur/runtimes/pull/4 when CI passes (except check-migration https://github.com/polkadot-fellows/runtimes/issues/200).

And I will merge this PR to `main` when `Encointer` bump is merged.

File: chain-spec-generator/Cargo.toml
Comment (ggwpez) [inline chain-spec-generator/Cargo.toml:44]: Can you open an issue for this please?

File: relay/polkadot/src/governance/old.rs
Comment (ggwpez) [inline relay/polkadot/src/governance/old.rs:1]: Was this file just unused?

**Review by ggwpez — APPROVED** [review]

File: relay/polkadot/src/governance/old.rs
Comment (bkontur) [reply relay/polkadot/src/governance/old.rs:1]: yes, exactly

File: system-parachains/encointer/src/weights/pallet_message_queue.rs
Comment (bkontur) [reply system-parachains/encointer/src/weights/pallet_message_queue.rs:22]: Yes, looks like it is just copied from other runtimes.
There are also others weight files copied from polkadot-sdk, but it is ok, because there is a point [here](https://github.com/polkadot-fellows/runtimes/issues/140 for regenerating all the weights for all runtimes before releasing 1.2.0.

File: chain-spec-generator/Cargo.toml
Comment (bkontur) [reply chain-spec-generator/Cargo.toml:44]: I am in the middle of the rebase of bump 1.7.0 https://github.com/polkadot-fellows/runtimes/pull/187, probably I will need it there, if not I will remove this.

But this could be a potential problem for next releases again and again. E.g. if we bump `polkadot-sdk` deps and Encointer has different sp-core version, it won't compile here. Once I am done with fixing conflicts and rebasing 1.7.0, I will check, if we could fix it somehow and avoid this hach, if not I will create issue.

Comment (brenzi) [reply chain-spec-generator/Cargo.toml:44]: We will need to find a way for Encointer's upgrades to be async with the other system chains. It is in Encointer's interest to stay up to date with other system chains and the sdk. I think we can commit to bumping our pallets within 2-3 weeks after a relevant polkadot-sdk upgrade meant to trigger a runtimes upgrade, but we might still need temporary workarounds

Comment (bkontur) [reply chain-spec-generator/Cargo.toml:44]: I think that this hack is just because of AuraConfig, but if we create function in encointer runtime `fn aura_config_for_chain_spec(seed) -> AuraConfig`, we won't need this hack, I will try that

File: system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs
Comment (bkchr) [inline system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:294]: Given the comment you have deleted, is this still okay to have here `0`?

File: relay/polkadot/src/lib.rs
Comment (bkchr) [reply relay/polkadot/src/lib.rs:784]: So we should use an easier origin here @joepetrowski?

File: chain-spec-generator/Cargo.toml
Comment (bkchr) [reply chain-spec-generator/Cargo.toml:44]: Yeah generally Encointer should be open to update their stuff async in their own speed.

File: relay/polkadot/src/lib.rs
Comment (joepetrowski) [reply relay/polkadot/src/lib.rs:784]: I was thinking that we leave it as Root now and then change to something like General Admin (or a new People Admin track) once it is on a parachain, since the feature will be frozen for some time during the migration to a parachain anyway. But if there is demand for this now then I don't have any objections to something easier.

File: system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs
Comment (joepetrowski) [reply system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:294]: Yeah this can be `0`

File: system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs
Comment (joepetrowski) [reply system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:363]: @muharem can confirm, but since all pools require KSM then a path length greater than 3 doesn't really make sense. If you want to go X to Y, then you go `X -> KSM -> Y`. There isn't any other possible intermediate step.

File: system-parachains/bridge-hubs/bridge-hub-kusama/primitives/src/lib.rs
Comment (bkontur) [reply system-parachains/bridge-hubs/bridge-hub-kusama/primitives/src/lib.rs:90]: Because [here](https://github.com/polkadot-fellows/runtimes/pull/159/commits/a0a61ae1e9613b6e9a6079dd9ad0ecbb341e89e3) I added improved sanity checks for these fees.

No, no, we won't change it on every release, only if those tests fail, which would mean that the fees can be actually lowered.

File: system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs
Comment (bkontur) [reply system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:363]: @bkchr yes, intentionally, based on discussions with @muharem  https://github.com/paritytech/polkadot-sdk/issues/3440

**Review by muharem — APPROVED** [review]

File: chain-spec-generator/Cargo.toml
Comment (bkontur) [reply chain-spec-generator/Cargo.toml]: I removed this hack for Encointer and fixed it, so we can bump Encointer independently.

https://github.com/polkadot-fellows/runtimes/pull/159/commits/bbf7c5bde3aad2b5b97b18744068e89d6925d9ae

File: CHANGELOG.md
Comment (bkontur) [reply CHANGELOG.md]: @bkchr what about this: https://github.com/polkadot-fellows/runtimes/pull/159/commits/5e336f68dba0cc83586f5be1a8221f09e497f577 ?

**Review by Ank4n — APPROVED** [review]

**Comment by bkontur** [comment]
looks like we should temporary skip `Required` for Encointer's `check-migrations` according to the https://github.com/polkadot-fellows/runtimes/issues/200#issuecomment-1964138542, until it is enacted live (or fix here with some migration)

**Pull request merged**
